crates/oxc_parser/src/lexer/trivia_builder.rs

```diff
@@ -201,6 +201,50 @@ impl TriviaBuilder {
     }
 }
 
+#[inline(always)]
+// BENCH > v6
+fn contains_license_or_preserve_comment(s: &str) -> bool {
+    use memchr::memchr_iter;
+    use std::ptr;
+
+    // Little-endian 8-byte constant for the ASCII string "preserve"
+    // spelled backwards because of little-endian load
+    const PRESERVE_LE: u64 = 0x6576_7265_7365_7270;
+    // Little-endian 8-byte constant for the ASCII string "license" (with a leading 0 byte)
+    const LICENSE_LE: u64 = 0x0065_736e_6563_696c;
+    // Mask to ignore the first (highest-order) byte when matching "license"
+    const LICENSE_MASK: u64 = 0x00FF_FFFF_FFFF_FFFF;
+
+    let hay = s.as_bytes();
+
+    if hay.len() < 9 {
+        return false;
+    }
+
+    let search_len = hay.len() - 8;
+
+    for i in memchr_iter(b'@', &hay[..search_len]) {
+        // SAFETY: We're about to read 8 bytes starting at hay[i+1],
+        // which is within bounds because i < search_len = len - 8
+        unsafe {
+            // `/** @preserve */`
+            //       ^
+            // Get a pointer to the byte directly after `@`
+            // this is ok, as we are using `read_unaligned` on the resulting pointer.
+            #[expect(clippy::cast_ptr_alignment)]
+            let ptr = hay.as_ptr().add(i + 1).cast::<u64>();
+            // Read the sunsequent 8 bytes.
+            let chunk = ptr::read_unaligned(ptr);
+            // Compare against "preserve" constant, or masked "license" constant
+            if chunk == PRESERVE_LE || (chunk & LICENSE_MASK) == LICENSE_LE {
+                return true;
+            }
```

Comment:
this is marginally faster (.2% from codespeed) than:

```
        let tail = &hay[i + 1..];

        if tail.starts_with(PAT1) || tail.starts_with(PAT2) {
            return true;
        }
```

so not sure if the added complexity is worth it